### ISSUE TITLE
emmylua-check doesn't pick up workspace libraries if they're ignored

### ISSUE THREAD

**Issue by mrcjkb** [EmmyLuaLs/emmylua-analyzer-rust#683]
### Given:

- A project with the following structure:

```
.
├── lua
│   └── my-app
│       └── init.lua
├── .luarc.json
├── .lux
│   ├── .gitignore
│   └── 5.1
│       ├── 478a726e64668e552b6e6f7afbbb55aea0ae2eba63fa75b2e0b9d1ebc1639fdb-argparse@0.7.1-1
```

- The following `.luarc.json`:

```json
{
  "workspace": {
    "library": [
      ".lux/5.1/478a726e64668e552b6e6f7afbbb55aea0ae2eba63fa75b2e0b9d1ebc1639fdb-argparse@0.7.1-1",
    ]
  }
}
```

- The following `--ignore`: `.lux/**/*`

And the following `lua/my-app/init.lua`:

```lua
local argparse = require("argparse")

local parser = argparse("script", "An example.")
parser:rgument("input", "Input file.")
     --^ This should be "argument"
```

### I would expect:

- emmylua_check to provide an error or a warning for the `parser:rgument("input", "Input file.")` line

### Actual behaviour:

```
No issues found
Check finished
```

If I remove the `--ignore`, emmlua_check outputs warnings and errors for files in the `.lux` directory (see also #635).

### Context:

I'm not actually using the CLI, I'm using it as a library in this PR:

- https://github.com/nvim-neorocks/lux/pull/849

My guess is that this may be a combination of two bugs:

1. The `--ignore` prevents the addition of the workspace libraries in the ignored directory.
2. Without the `--ignore`, because it's in the project root, emmylua_check treats it as part of the workspace.

**Comment by mrcjkb** [comment]
Thanks for the quick response. 🙏 

I've tried pinning the latest commit (0e11e22d9cf31612bb93695c0efb97133ba49cb2 as of writing) and I'm still getting the same behaviour.

Cargo.lock entry:

```toml
[[package]]
name = "emmylua_check"
version = "0.10.0"
source = "git+https://github.com/EmmyLuaLs/emmylua-analyzer-rust.git?rev=0e11e22d9cf31612bb93695c0efb97133ba49cb2#0e11e22d9cf31612bb93695c0efb97133ba49cb2"
```

Edit: Also with commit 95c287a.

Could it be that the libs directory being in a subdirectory of the project root makes emmylua_check treat it as part of the main workspace?
